### PR TITLE
Module3 task done | fix defects

### PR DESCRIPTION
**Pull request by never47**

### CODE REVIEW

File: src/main/java/springTask/Utils.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            break;
+        }
+
+        return counter;
+    }
+
+    public static String usernameGen(String name, String surname, int number){
+        StringBuilder currUsername = new StringBuilder();
+        currUsername.append(name).append(".").append(surname);
```
Comment (HannaVatlina) [inline src/main/java/springTask/Utils.java:39]: line 16 (StringBuilder currNickname = new StringBuilder(firstName + "." + lastName);) and this line are the same operation?

File: src/test/java/springTask/service/TraineeServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import springTask.dao.TraineeRepository;
+import springTask.entity.Trainee;
+
+import java.util.List;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TraineeServiceTest {
```
Comment (HannaVatlina) [inline src/test/java/springTask/service/TraineeServiceTest.java:20]: It is good practice to test not only positive scenarios, but negative ones too.

File: src/main/java/springTask/dao/TraineeRepository.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public void create(Trainee trainee){
+        dataBase.put(trainee.getUsername(), trainee);
+    }
+
+    public Trainee selectByUsername(String key){
+        if(!dataBase.containsKey(key)){
+            throw  new IllegalArgumentException("no trainee with such key");
+        }
```
[the remark this answers is not in the document]
Comment (never47) [reply src/main/java/springTask/dao/TraineeRepository.java:68]: done

File: src/main/java/springTask/service/ServiceFacade.java [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+        this.traineeService = traineeService;
+        this.trainerService = trainerService;
+        this.trainingService = trainingService;
+    }
+    
+    public void run(){
+        try(Scanner scanner = new Scanner(System.in)){
+           while(isAppRunning){
+               if(!isLogged) loggedOutState(scanner);
+               else loggedInState(scanner);
```
[the remark this answers is not in the document]
Comment (never47) [reply src/main/java/springTask/service/ServiceFacade.java:36]: done

File: src/main/java/springTask/service/ServiceFacade.java [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+        System.out.println("Enter your password");
+        String pass = scanner.nextLine();
+
+        try{
+            Trainee trainee = traineeService.selectTrainee(login);
+            if(pass.equals(trainee.getPassword())){
+                isLogged = true;
+            }
+        }catch (IllegalArgumentException e){
+            //
```
[the remark this answers is not in the document]
Comment (never47) [reply src/main/java/springTask/service/ServiceFacade.java:91]: I think printing something like - That is not a trainee/trainer is redundant for logging, I use this try catch blocks to check if entered username exists in traineeRepository or trainerRepository...

File: src/main/java/springTask/Utils.java (hunk 1 above)
Comment (never47) [reply src/main/java/springTask/Utils.java:39]: No, method countUsernameSerialNumber() just counts the next serialNumber for new nickname, it checks that if there exists for example hanna.vatlina in database, that means that the next one must have serialNumber equal to 1 -> hanna.vatlina1

Then the other function usernameGen(String name, String surname, int number) uses this value to generate new username

My fault, countUsernameSerialNumber method must be private and I should call it from usernameGen, changed it

**Pull request closed without merging**